### PR TITLE
Closes #2338: keep_nas = TRUE example in derive_param_computed() documentation

### LINKED ISSUE #2338: Documentation: unable to derive a new variable using derive_param_computed() when source variable is NA
### What happened?

I'd like use use derive_param_computed() to create a new parameter for a BDS ADAM (e.g. ADVS). I also wanna directly copy over some source variables to the new parameter, such as ADT, ADY, ADTF, ATMF etc. When the source variables are non-missing, this step is working fine. However, when the source variables are NA for all, the new parameter could not be created properly.

### Session Information

R version 4.3.1 (2023-06-16)
admiral_0.12.2

### Reproducible Example
```
# Create a dummy dataset for testing
advs <- tribble(
  ~USUBJID,      ~PARAMCD, ~PARAM,                            ~AVAL, ~AVALU, ~VISIT,
  "01-701-1015", "DIABP",  "Diastolic Blood Pressure (mmHg)",    51, "mmHg", "BASELINE",
  "01-701-1015", "DIABP",  "Diastolic Blood Pressure (mmHg)",    50, "mmHg", "WEEK 2",
  "01-701-1015", "SYSBP",  "Systolic Blood Pressure (mmHg)",    121, "mmHg", "BASELINE",
  "01-701-1015", "SYSBP",  "Systolic Blood Pressure (mmHg)",    121, "mmHg", "WEEK 2",
  "01-701-1028", "DIABP",  "Diastolic Blood Pressure (mmHg)",    79, "mmHg", "BASELINE",
  "01-701-1028", "DIABP",  "Diastolic Blood Pressure (mmHg)",    80, "mmHg", "WEEK 2",
  "01-701-1028", "SYSBP",  "Systolic Blood Pressure (mmHg)",    130, "mmHg", "BASELINE",
  "01-701-1028", "SYSBP",  "Systolic Blood Pressure (mmHg)",    132, "mmHg", "WEEK 2"
)

# Derive a variables with all NAs
advs$ADTF <- NA_character_

# Examples to deriving a new parameter with all source vairables non-missing
# You can see it derives the new parameter "MAP" properly
derive_param_computed(
  advs,
  by_vars = exprs(USUBJID, VISIT),
  parameters = c("SYSBP", "DIABP"),
  set_values_to = exprs(
    AVAL = (AVAL.SYSBP + 2 * AVAL.DIABP) / 3,
    PARAMCD = "MAP",
    PARAM = "Mean Arterial Pressure (mmHg)",
    AVALU = "mmHg"
  )
)

# Examples to deriving a new parameter with one all-NA variable ADTF
# You can see it didn't derive the new parameter "MAP"
derive_param_computed(
  advs,
  by_vars = exprs(USUBJID, VISIT),
  parameters = c("SYSBP", "DIABP"),
  set_values_to = exprs(
    AVAL = (AVAL.SYSBP + 2 * AVAL.DIABP) / 3,
    PARAMCD = "MAP",
    PARAM = "Mean Arterial Pressure (mmHg)",
    AVALU = "mmHg",
    ADTF = ADTF.SYSBP
  )
)
```

### PR DESCRIPTION
**Pull request by manciniedoardo**
Closes #2338 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
@bms63 @bundfussr I added the example by adding `ADT` and `ADTF` variables to the `advs` object. I did it on a new line to avoid overflowing past 100 chars in the tibble, not sure if this is the best approach though as it's not super neat?

@w316liu please feel free to take a look as well!

File: man/derive_param_computed.Rd [hunk 1]
```diff
@@ -179,10 +186,29 @@ derive_param_computed(
...
     PARAM = "Mean Arterial Pressure (mmHg)",
-    AVALU = "mmHg"
+    AVALU = "mmHg",
+    ADT = ADT.SYSBP
   )
 )
 
+Example 1b: Using option `keep_nas = TRUE` to derive MAP in the case where some/all values
+of a variable used in the computation are missing
+
```
Comment (bms63) [inline man/derive_param_computed.Rd:196]: Is there data that has NAs in it?  I don't see it here?

Comment (manciniedoardo) [reply man/derive_param_computed.Rd:196]: `ADTF` is all `NA` and it's one of the variables inside `set_values_to` in the second example (1b)

Comment (bms63) [reply man/derive_param_computed.Rd:196]: oh right - okay I thought it was the specific calculation for doing MAP that needed NAs, but looking at issue I see I was misunderstanding.  

I think you need to add a # in your comment for 1b for the checks to pass.

Comment (manciniedoardo) [reply man/derive_param_computed.Rd:196]: ah yes of course, thanks

**Comment by bms63** [comment]
The error is still persisting :(

**Comment by manciniedoardo** [comment]
> The error is still persisting :(

fixed now

**Pull request merged**